addons/website/static/tests/tours/grid_layout.js

```diff
@@ -43,3 +43,33 @@ wTourUtils.registerWebsitePreviewTour('website_replace_grid_image', {
     },
     ...wTourUtils.clickOnSave()
 ]);
+
+wTourUtils.registerWebsitePreviewTour("scroll_to_new_grid_item", {
+    test: true,
+    url: "/",
+    edition: true,
+}, [
+    // Drop enough snippets to scroll.
+    wTourUtils.dragNDrop({id: "s_text_image", name: "Text - Image"}),
+    wTourUtils.dragNDrop({id: "s_image_text", name: "Image - Text"}),
+    wTourUtils.dragNDrop({id: "s_image_text", name: "Image - Text"}),
+    // Toggle the first snippet to grid mode.
+    wTourUtils.clickOnSnippet({id: "s_text_image", name: "Text - Image"}),
+    wTourUtils.changeOption("layout_column", 'we-button[data-name="grid_mode"]'),
+    // Add a new grid item.
+    wTourUtils.changeOption("layout_column", 'we-button[data-add-element="image"]'),
+    {
+        content: "Check that the screen scrolled to the new grid item",
+        trigger: "iframe .s_text_image .o_grid_item:nth-child(3)",
+        run: function () {
+            // Leave some time to the page to scroll.
+            setTimeout(() => {
+                const newItemPosition = this.$anchor[0].getBoundingClientRect();
+                if (newItemPosition.top < 0) {
+                    console.error("The page did not scroll to the new grid item.");
+                }
+            }, 500);
+        },
+    },
+    ...wTourUtils.clickOnSave(),
```

Comment:
I think there is something wrong here. Why do you need to click on save button ? Is it because save takes more than 500 ms ? I thing the `Check that the screen scrolled to the new grid item` step should add a class to the body element after the check `if (newItemPosition.top < 0) {` and your last step should check that the class in on the body. That way, you are sure that your check in the setTimeout is done before finishing the tour.

Reply:
it's always a good practice that your last step is a check and not an action :+1: